pkg/v1/tkg/test/tkgctl/shared/e2e_common.go

```diff
@@ -150,6 +155,41 @@ func E2ECommonSpec(context context.Context, inputGetter func() E2ECommonSpecInpu
 		By(fmt.Sprintf("Waiting for workload cluster %q nodes to be up and running", clusterName))
 		framework.WaitForNodes(framework.NewClusterProxy(clusterName, tempFilePath, ""), 2)
 
+		if input.Plan == "devcc" {
+			By(fmt.Sprintf("Generating credentials for management cluster %q", input.E2EConfig.ManagementClusterName))
+			mngkubeConfigFileName := input.E2EConfig.ManagementClusterName + ".kubeconfig"
+			mngtempFilePath := filepath.Join(os.TempDir(), mngkubeConfigFileName)
+			err = tkgCtlClient.GetCredentials(tkgctl.GetWorkloadClusterCredentialsOptions{
+				ClusterName: input.E2EConfig.ManagementClusterName,
+				Namespace:   "tkg-system",
+				ExportFile:  mngtempFilePath,
+			})
+			Expect(err).To(BeNil())
+
+			mngscheme := runtime.NewScheme()
+			err = pkgiv1alpha1.AddToScheme(mngscheme)
+			Expect(err).NotTo(HaveOccurred())
+			err = runtanzuv1alpha3.AddToScheme(mngscheme)
+			Expect(err).NotTo(HaveOccurred())
+
+			// create k8sclient for management cluster
+			mngclient, err := createClientFromKubeconfig(mngtempFilePath, mngscheme)
```

Comment:
mngclient is declared inside the if loop. This makes it unaccessible to anything outside the if block. 
It might be a good idea to declare the varaible outside of if statement. ``` var mngclient client.Client ``` before the if loop. That way the client only needs to be created once.